### PR TITLE
Feature/study form

### PR DESCRIPTION
**Pull request by eleanorgraham**
In anticipation of our 11/12/19 meeting, I am hoping to get some new enhancements onto staging for the group to test:

This pull request includes a functioning Study Add/Edit/Delete form. There are buttons on the study detail pages to get to the edit, and a button on the main studies page to get to the add. While the functions in a basic way right now, there are a few missing pieces to be added: Investigators, Image upload, and the Topic multi-select tree. 

I have also included a little addition of date to User Admin to aid in sorting new users. I have not fleshed out the user date tracking that we've outlined, just surfaced the data that we already had to aid the team while they are waiting for more features.

### CODE REVIEW

**Comment by eleanorgraham** [comment]
Sometimes when adding a new study it will save two copies of itself. I can't consistently reproduce it and I thought I'd fixed it, but it happened with the first save on staging... So I'll need to look at that more closely before we merge this.

**Comment by eleanorgraham** [comment]
Also discovered during the demo that the edit function is misbehaving.

File: frontend/src/app/resource-form/resource-form.component.ts [hunk 1]
```diff
@@ -323,38 +323,29 @@ export class ResourceFormComponent implements OnInit {
...
     if (this.form.valid) {
       if (this.createNew) {
-        if (this.model.type == 'resource') {
-          this.addAndClose(this.api.addResource(this.model));
-        } else if (this.model.type == 'location') {
-          this.addAndClose(this.api.addLocation(this.model));
-        } else if (this.model.type == 'event') {
-          this.addAndClose(this.api.addEvent(this.model));
-        }
+        this.updateOrganization(() => this.addAndClose(this.api[`add${resourceType}`](this.model)));
```
Comment (danfunk) [inline frontend/src/app/resource-form/resource-form.component.ts]: VASTLY Cleaner.

File: frontend/src/app/search/search.component.html [hunk 2]
```diff
@@ -59,7 +59,7 @@ <h1 *ngIf="selectedType" fragment="filter-top">
     <h1 *ngIf="!selectedType" fragment="filter-top">
       All Resources
     </h1>
-    <app-resource-add-button [currentUser]="currentUser"></app-resource-add-button>
```
Comment (danfunk) [inline frontend/src/app/search/search.component.html]: Also much clearer and tighter - good reuse.

File: frontend/src/app/study-form/study-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+  study: Study;
+  updatedStudy: Study;
+  pageState = PageState;
+  state = PageState.LOADING;
+  showConfirmDelete = false;
+
+  model: any = {};
+  updatedModel: any = {};
+  form: FormGroup;
+  fields: FormlyFieldConfig[] = [
```
Comment (danfunk) [inline frontend/src/app/study-form/study-form.component.ts]: Looking forward to seeing this go away!!!

**Pull request merged**